pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -477,6 +477,15 @@ func containerSucceeded(c *v1.Container, podStatus *kubecontainer.PodStatus) boo
 func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *kubecontainer.PodStatus) podActions {
 	klog.V(5).Infof("Syncing Pod %q: %+v", format.Pod(pod), pod)
 
+	// All containers in the pod are done and the pod may be cleaned up
+	if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
```

Comment:
this seems to match the GC conditions
https://github.com/kubernetes/kubernetes/blob/90237ce89a7c252c44162b841a37d85c7d2ef71f/pkg/controller/podgc/gc_controller.go#L120-L125

Reply:
@aojea thanks for the double-check!